context/switch-arm64.c

```diff
@@ -105,7 +105,12 @@ FSIZE(switch_frame)
 FNAME(create_frame)
 "       stp x1, x2, [x0, #-16]!\n"
 "       mov x16, lr\n"               // Save LR so can use same STP slot
+#if defined(__aarch64__)  &&  defined(__APPLE__) // darwin syntax
+"       adrp lr, action_entry@PAGE\n"
+"       add lr, lr, action_entry@PAGEOFF\n"
+#else
 "       ldr lr, =action_entry\n"
+#endif
```

Comment:
I don't think there's anything specific to Apple about this two part syntax.  If possible I think I'd like to see this unconditionally using the new syntax.

It does occur to me that we shouldn't need to be using a linker relocation here at all: `action_entry` is nearby at a constant offset, so I'd expect something like
```asm
 adr lr, action_entry
```
to work on both targets (it looks like [ADR](https://developer.arm.com/documentation/ddi0602/2025-06/Base-Instructions/ADR--Form-PC-relative-address-) is the appropriate instruction here.

@AlexanderWells-diamond, do we have CI runners for Linux and OSX ARM64 for cothread?

Reply:
We currently do not have Linux ARM64, nor any MacOS CI as part of our default set.

In theory, as per [this blog post](https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/), we can simply add `ubuntu-24.04-arm` and `macos-latest` to the test matrix to get full coverage of Arm64 platforms. However, a [quick test](https://github.com/DiamondLightSource/cothread/actions/runs/17726096608) shows failures on both platforms.

I'll do a little digging into why these are failing.

Reply:
Indeed, the corresponding instruction in `switch-arm.c` also ought to be [ADR](https://developer.arm.com/documentation/dui0379/e/arm-and-thumb-instructions/adr--pc-relative-), but I guess that's out of scope for this PR.